### ISSUE TITLE
v1.3.4 UI Regression (possible state management regression)

### ISSUE THREAD

**Issue by djenriquez** [hashicorp/nomad#14412]
### Nomad version
```sh
nomad version
Nomad v1.3.4 (a267b2b000ae72ba05fa571eaafe170a7b9c4d1b)
```

### Operating system and Environment details
Amazon Linux 2

### Issue
After upgrading from 1.3.3 to 1.3.4, seeing some regression in the UI:
1. The server page is not properly identifying the leader (every node has leader:false even though the /v1/status/leader endpoint reveals the leader), nor is it clearing out the old 1.3.3 servers that have left even after 24 hours (the data from `v1/agent/members` does not seem to be clearing out the old servers)
2. In our development environment where ACLs are disabled, the "exec" button is disabled, even though you can click this button and an exec pop up comes up.
![Screen Shot 2022-08-30 at 2 10 52 PM](https://user-images.githubusercontent.com/4636740/187543883-fcb4f75b-cf03-435e-b49a-0c2514a21388.png)
3. Related to 2, after the exec pop up window comes up, you are unable to actually run the command to execute the exec. The window just hangs. Chrome dev tools show no network traffic after pressing enter, its as if its just disabled completely.
![Screen Shot 2022-08-30 at 2 12 31 PM](https://user-images.githubusercontent.com/4636740/187544211-cce35b6f-8663-460c-ab79-545cf2125e6c.png)

4. Also seeing a js error in the console (hard to tell if these are expected since JS errors are so common these days):
```
Failed to load resource: the server responded with a status of 400 ()

nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:2076 Uncaught TypeError: Cannot read properties of null (reading 'includes')
    at get activeRegion (nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:2076:10)
    at vendor-eb07b60f7e1f309e783749d1ba18f9d7.js:2000:67
    at e.untrack (vendor-eb07b60f7e1f309e783749d1ba18f9d7.js:4318:12)
    at Re.get (vendor-eb07b60f7e1f309e783749d1ba18f9d7.js:2000:56)
    at z.r [as activeRegion] (vendor-eb07b60f7e1f309e783749d1ba18f9d7.js:1964:245)
    at r.getTaskStateSocket (nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:2051:179)
    at h.openAndConnectSocket (nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:1141:844)
    at e.default.handleDataEvent (nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:2493:60)
    at nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:2492:130
    at e.fire (chunk.331.2bf16a8aab827bcd7550.js:786:29)
get activeRegion @ nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:2076
(anonymous) @ vendor-eb07b60f7e1f309e783749d1ba18f9d7.js:2000
e.untrack @ vendor-eb07b60f7e1f309e783749d1ba18f9d7.js:4318
get @ vendor-eb07b60f7e1f309e783749d1ba18f9d7.js:2000
r @ vendor-eb07b60f7e1f309e783749d1ba18f9d7.js:1964
getTaskStateSocket @ nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:2051
openAndConnectSocket @ nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:1141
handleDataEvent @ nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:2493
(anonymous) @ nomad-ui-2337cfef252f8cd23727e6c21be6e3d8.js:2492
e.fire @ chunk.331.2bf16a8aab827bcd7550.js:786
(anonymous) @ chunk.331.2bf16a8aab827bcd7550.js:787
e.fire @ chunk.331.2bf16a8aab827bcd7550.js:786
t.triggerDataEvent @ chunk.331.2bf16a8aab827bcd7550.js:1502
t._keyDown @ chunk.331.2bf16a8aab827bcd7550.js:221
(anonymous) @ chunk.331.2bf16a8aab827bcd7550.js:181

<REDACTED>:1          Failed to load resource: the server responded with a status of 500 ()

allocations:1          Failed to load resource: the server responded with a status of 500 ()
```

5.  When loading up the jobs page:
![Screen Shot 2022-08-30 at 2 15 09 PM](https://user-images.githubusercontent.com/4636740/187544661-63d50f5c-45ff-46e4-a5ed-812aa32c50a7.png)

6. The UI no longer displays the Nomad Version on the bottom left hand corner (bottom of the navigation menu).
7. Clients that have rotated out are no longer being GC'd, atleast not as the same rate as before, hundreds of clients report as `down` (we have a busy cluster). This screenshot is filtering "Down" status.
![Screen Shot 2022-08-30 at 2 19 57 PM](https://user-images.githubusercontent.com/4636740/187545391-4329054a-4a1b-47a9-aefc-1532519016b9.png)

### Reproduction steps

Upgrade Nomad from 1.3.3 to 1.3.4.

**Comment by djenriquez** [comment]
I found other similar issues after posting, I do have some cases not discussed in those issues though. Should I keep this issue open for that or just close and post the additional info over on the existing issues?

**Comment by aniaptebsft** [comment]
Seeing the same issue with grayed out exec button. This is causing a lot of disruption to our engineering workflows.

**Comment by jrasell** [comment]
Hi @djenriquez and @aniaptebsft and thanks for raising this with the context. I will link the issue to https://github.com/hashicorp/nomad/issues/14359 which has been tracking some UI regressions within 1.3.4, however, I will let @philrenaud read through this and decided whether to close this issue out as duplicate.

**Comment by shantanugadgil** [comment]
Hi, we are seeing a bunch of the UI issues reported after upgrading to 1.3.4 for the Nomad servers.

This particular issue (`Exec` button disabled) is especially bothersome.

The UI shows the version a the bottom left only on the first load, and then it disappears.

cross referencing #14359   #14331

**Comment by shantanugadgil** [comment]
Another question @jrasell @philrenaud how concerned should we be with version 1.3.4 and the non-UI functionality. Have there been any issues reported which are *not related* to the GUI?

The current state is that we have upgraded ONLY the servers to 1.3.4 and were planning to roll out the clients with 1.3.4 within the week.

**Comment by philrenaud** [comment]
> Have there been any issues reported which are not related to the GUI?

Hi @shantanugadgil — the issues described are solved in https://github.com/hashicorp/nomad/pull/14381 which are restricted to the UI alone. We would still recommend moving to 1.3.5 when that releases (very shortly)

**Comment by philrenaud** [comment]
Hi @djenriquez , @aniaptebsft — sorry for the hassle here. The issues you see will be fixed with 1.3.5, going out shortly, and can temporarily be alleviated on a per-browser basis by adding the following to your browser console:

`localStorage.setItem('nomadTokenSecret', 'thisisathirtysixcharacterlongstring!');` (per [this comment](https://github.com/hashicorp/nomad/issues/14359#issuecomment-1229367069))

Closing this issue and tracking in #14359

**Comment by shantanugadgil** [comment]
This is good news indeed. Thanks! 👍🏻🙏🏻
